### PR TITLE
[1.1.x] Add `apiArtifactsMountEmptyDir` and `dropinsMountEmptyDir`configs

### PR DESCRIPTION
**Pull request by NomadXD**
## Purpose
Add `apiArtifactsMountEmptyDir` and `dropinsMountEmptyDir`configs

Fixes: #62 #64

### CODE REVIEW

File: helm/choreo-connect/README.md [hunk 1]
```diff
@@ -394,6 +394,7 @@ If you do not have active WSO2 subscription do not change the parameters `wso2.d
 | `wso2.deployment.adapter.podAnnotations`                                    | Key value pair of annotations for the pod                                                 | sidecar.istio.io/inject: "false" |
 | `wso2.deployment.adapter.configToml`                                        | Define templated config.toml file, if empty using default config.toml                     | Default templated config toml file |
 | `wso2.deployment.adapter.logConfigToml`                                     | Define templated log_config.toml file, if empty using default log_config.toml             | Default templated log config toml file |
+| `wso2.deployment.adapter.apiArtifactsMountEmptyDir`         | Mount API artifacts directory if the directory is empty         |           false         |
```
Comment (renuka-fernando) [inline helm/choreo-connect/README.md:397]:
```suggestion
| `wso2.deployment.adapter.apiArtifactsMountEmptyDir`         | Mount an empty directory on API artifacts directory         |           false         |
```

File: helm/choreo-connect/README.md [hunk 2]
```diff
@@ -430,6 +431,7 @@ Gateway runtime (enforcer + router) deployment configurations
 | `wso2.deployment.gatewayRuntime.enforcer.imagePullPolicy`                   | Image pull policy of the container                                                        | "IfNotPresent"              |
 | `wso2.deployment.gatewayRuntime.enforcer.envOverride`                       | Set (or override) environment variables as values or from ConfigMaps or Secrets           | Default Passwords           |
 | `wso2.deployment.gatewayRuntime.enforcer.dropins`                           | Mount enforcer lib dropins JARs to the `dropins` directory, array of ConfigMap names      | -                           |
+| `wso2.deployment.gatewayRuntime.enforcer.dropinsMountEmptyDir`          | Mount dropins JARs directory if the directory is empty          |           true          |
```
Comment (renuka-fernando) [inline helm/choreo-connect/README.md:434]:
```suggestion
| `wso2.deployment.gatewayRuntime.enforcer.dropinsMountEmptyDir`          | Mount an empty directory on dropins JARs directory          |           true          |
```

File: helm/choreo-connect/templates/gateway-runtime-deployment.yaml [hunk 3]
```diff
@@ -113,8 +113,10 @@ spec:
             - mountPath: /home/wso2/check_health.sh
               name: grpc-probe-health-check
               subPath: "check_health.sh"
+          {{- if or (.Values.wso2.deployment.gatewayRuntime.enforcer.dropins) (.Values.wso2.deployment.gatewayRuntime.enforcer.dropinsMountEmptyDir) }}
             - mountPath: /home/wso2/lib/dropins
               name: enforcer-dropins
```
Comment (renuka-fernando) [inline helm/choreo-connect/templates/gateway-runtime-deployment.yaml:118]: Shall we remove the Volume for `enforcer-dropins` as well? if the condition not met

File: helm/choreo-connect/values.yaml [hunk 4]
```diff
@@ -148,6 +148,9 @@ wso2:
       # Define templated log_config.toml file, if empty using default log_config.toml
       logConfigToml: ""
 
+      # Mount API artifacts directory if the directory is empty
+      apiArtifactsMountEmptyDir: false
```
Comment (pubudu538) [inline helm/choreo-connect/values.yaml:152]: Default should be true right? @renuka-fernando

File: helm/choreo-connect/values.yaml [hunk 5]
```diff
@@ -148,6 +148,9 @@ wso2:
       # Define templated log_config.toml file, if empty using default log_config.toml
       logConfigToml: ""
 
+      # Mount API artifacts directory if the directory is empty
```
Comment (renuka-fernando) [inline helm/choreo-connect/values.yaml:151]: Lets update these comments same as Readme file.

File: helm/choreo-connect/values.yaml (hunk 4 above)
Comment (renuka-fernando) [inline helm/choreo-connect/values.yaml:152]:
```suggestion
      apiArtifactsMountEmptyDir: true
```

Comment (renuka-fernando) [reply helm/choreo-connect/values.yaml:152]: Let's keep true as default

Comment (renuka-fernando) [reply helm/choreo-connect/values.yaml:152]: yes

File: helm/choreo-connect/templates/gateway-runtime-deployment.yaml (hunk 3 above)
Comment (NomadXD) [reply helm/choreo-connect/templates/gateway-runtime-deployment.yaml:118]: fixed

File: helm/choreo-connect/values.yaml (hunk 4 above)
Comment (NomadXD) [reply helm/choreo-connect/values.yaml]: fixed

File: helm/choreo-connect/values.yaml (hunk 5 above)
Comment (NomadXD) [reply helm/choreo-connect/values.yaml]: fixed

File: helm/choreo-connect/values.yaml (hunk 4 above)
Comment (NomadXD) [reply helm/choreo-connect/values.yaml]: @pubudu538 @renuka-fernando If the default config is `true`, do we need to add `RUN mkdir -p /home/wso2/artifacts/apis` to the Dockerfile of the adaper ?

**Review by renuka-fernando — APPROVED** [review]

File: helm/choreo-connect/values.yaml (hunk 4 above)
Comment (renuka-fernando) [reply helm/choreo-connect/values.yaml]: We will be adding this `RUN mkdir -p /home/wso2/artifacts/apis` in the next release of Choreo Connect. In Helm release 1.2.0.0 and these changes will be reverted.
Created an issue to track this https://github.com/wso2/kubernetes-microgateway/issues/61

**Review by chirangaalwis — APPROVED** [review]

**Pull request merged**